cluster-default-publishing-strategy.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Cluster Publishing Strategy
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - "@openshift/api-reviewers"
+  - "@openshift/openshift-team-installer"
+  - "@ironcladlou"
+approvers:
+  - "@openshift/api-approvers"
+creation-date: 2019-09-18
+last-updated: 2019-09-18
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Cluster Publishing Strategy
+
+Administrators would like to be able to install OpenShift cluster as `internal/private`, which is only accessible from within the network, VPN tunnels and, not visible to the Internet. Therefore all the cluster operators that create endpoints for the cluster would need a place to discover this intent of the administrator and ensure that their resources are only accessible from within the network, VPN tunnels and not visible to the Internet by default.
+
+This should not block administrators from using operator-level configuration to override publishing strategy of individual endpoints.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+See the [section][#cluster-publishing-strategy]
+
+## Motivation
+
+See the [section][#cluster-publishing-strategy]
+
+### Goals
+
+1. Define an API that allows all the cluster operators to discover the default cluster publishing strategy.
+
+### Non-Goals
+
+## Proposal
+
+Modify the existing `config.openshift.io`'s custom resource [`Infrastructure`][infrastructure-api] to include an enum field `Publish` that can be set to `External` or `Internal`.
+
+The field will be added to [`status` section][infrastructure-api-status] because the goal is to provide an API for discovery i.e. the current state, and not to provide an API for customers to specify the desired state.
+
+### User Stories
+
+#### Default Ingress Controller LoadBalancer Scope
+
+The ingress-operator will be able to use this field to define the **default** [scope][ingress-controller-api-load-balancer-scope] for the LoadBalancers for the [ingress controllers][ingress-controller-api].
```

Comment:
@ironcladlou is this related to your designs?

Reply:
https://github.com/openshift/enhancements/pull/31 is "bring your own IngressController", and this PR is (I think) about adding some new focused API so that you don't have to bring your own (usually, and for this specific use case).